dipy/workflows/tracking.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+from __future__ import division
+
+import logging
+import numpy as np
+
+from nibabel.streamlines import save, Tractogram
+
+from dipy.direction import DeterministicMaximumDirectionGetter
+from dipy.io.image import load_nifti
+from dipy.io.peaks import load_peaks
+from dipy.tracking import utils
+from dipy.tracking.local import (ThresholdTissueClassifier,
+                                 LocalTracking)
+from dipy.workflows.workflow import Workflow
+
+
+class GenericTrackFlow(Workflow):
+    def _core_run(self, stopping_path, stopping_thr, seeding_path,
+                  seed_density, use_sh, pam, out_tract):
+
+        stop, affine = load_nifti(stopping_path)
+        classifier = ThresholdTissueClassifier(stop, stopping_thr)
+        logging.info('classifier done')
+        seed_mask, _ = load_nifti(seeding_path)
+        seeds = \
+            utils.seeds_from_mask(
+                seed_mask,
+                density=[seed_density, seed_density, seed_density],
+                affine=affine)
+        logging.info('seeds done')
+        direction_getter = pam
+
+        if use_sh:
+            direction_getter = \
+                DeterministicMaximumDirectionGetter.from_shcoeff(
+                    pam.shm_coeff,
+                    max_angle=30.,
+                    sphere=pam.sphere)
+
+        streamlines = LocalTracking(direction_getter, classifier,
+                                    seeds, affine, step_size=.5)
+        logging.info('LocalTracking initiated')
+
+        tractogram = Tractogram(streamlines, affine_to_rasmm=np.eye(4))
+        save(tractogram, out_tract)
+
+        logging.info('Saved {0}'.format(out_tract))
+
+
+class DetTrackPAMFlow(GenericTrackFlow):
+    @classmethod
+    def get_short_name(cls):
+        return 'det_track'
+
+    def run(self, pam_files, stopping_files, seeding_files,
+            stopping_thr=0.2,
+            seed_density=1,
+            use_sh=False,
+            out_dir='',
+            out_tractogram='tractogram.trk'):
+
+        """ Workflow for deterministic using a saved peaks and metrics (PAM)
+        file as input.
+
+        Parameters
+        ----------
+        pam_files : string
+           Path to the peaks and metrics files. This path may contain
+           wildcards to use multiple masks at once.
```

Comment:
In my bash terminal, there is no space between the last word of a line and the first of the next line. e.g. for line 69-70 I have `... may containwildcards to use ....`